enhancements/installer/log-config.md

```diff
@@ -0,0 +1,189 @@
+
+This design proposes we add support for a log-config.yaml file to
+allow users to pass in logging configuration, for example:
+
+```go
+type LogConfig struct {
+  // Fields contains user-provided fields to be added to all log entries.
+  // +optional
+  Fields map[string]string `json:"fields,omitempty"`
+
+  // Level sets the level of logging to standard out. 
+  // Valid values: Info, Debug, Warn, Error
+  // +optional
+  Level string `json:"level,omitempty"`
+}
+```
+
+The install-config is the typical vehicle for installer configuration,
+but it is not acceptable in this use case because the install-config
+is an Installer Asset, and logging must be configured:
+
+1. Before assets are initialized
+2. Despite asset failures
+
+In the example use case of decorating logs with fields, we need
+to decorate lines that are output before the install config is loaded
+and we need to decorate output that would occur if there are
+errors in the install config and it cannot be loaded.
+
+The log-config.yaml file would be loaded if present in the install directory,
+as well as adding a command line flag  --log-config  to accept a path to the config file.
+
+Users should be able to provide their config in a file that looks, for example, like:
+
+```yaml
+fields:
+  key1: value1
+  key2: value2
+  key3: value3
+level: debug
+```
+
+### Workflow Description
+
+1. Installer user specifies configuration in `log-config.yaml`
+2. `log-config.yaml` is placed in install dir or path is specified with a flag
+3. Installer uses `log-config.yaml` for any Installer commands
+
+
+### API Extensions
+
+n/a
+
+### Risks and Mitigations
+
+Adding the ability to configure logging comes with a risk that the
+logging could have errors or misconfiguration. There is an Open
+Question about whether logging errors should fail the install or
+just log a warning message:
+
+If users include a `log-config.yaml` file, they intend to configure
+logging. If there is an error in that file, should we prevent installs
+until that configuration is properly handled? Or should we make the
+best effort to complete the install although the logging is not
+properly configured? 
+
+My weakly held opinion is that we should prevent attempting installs if
+`log-config.yaml` is supplied but is misconfigured.
+
+### Drawbacks
+
+This enhancement proposes a new way to introduce logging configuration.
+The existing methods we have are: install config, CLI flags, and 
+environment variables (not supported for customer user). By introducing
+another method of configuration, we risk confusion. I do not think
+this is a serious drawback or perhaps even a valid one.
+
+## Design Details
+
+### Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ 1. Should logging misconfiguration prevent installs? (see Risks & Mitigations)
+ 2. Should log-config.yaml be versioned according to k8s API versioning?
```

Comment:
Mm, this might be a good idea for the compat reasons I mentioned above. That way installer could break compatibility with a new version, as long as the old version continues to be supported until the respective OCP version is EOL.

(NB: It doesn't have to be k8s style versioning; could be as simple as a `version` field in the config format.)